### ISSUE TITLE
Problems fine-tuning on industrial dataset

### ISSUE THREAD

**Issue by giorgiacarbonififth-cmyk** [manycore-research/SpatialLM#108]
Thank you for your amazing work, this model is truly impressive. 

I was trying to fine-tune the model on my custom dataset consisting of ply scans of an electric power plant (classes include valves, pipes, electric equipment and so on). I scaled, transformed and rotated each scene as required. Each room is around 3m to 33.6m height. I am training using LLama 1B, 1e-04 lr, 1280 bins, unfreezed model, batch size of 2, 20 gradient accumulation steps, 45 epochs. Loss decreases steadily during training and eval.

Problem: it gets 0 correct bounding boxes. It learns the classes, but can't place the boxes. Inference result (the result is not consistent, it changes at every run on the same scene):

bbox_0=Bbox(civils,0,6,5,241,7,0,0)
bbox_1=Bbox(civils,0,6,5,241,7,0,0)
bbox_2=Bbox(civils,0,6,5,241,7,0,0)
bbox_3=Bbox(civils,0,6,5,241,7,0,0)
bbox_4=Bbox(civils,0,6,5,241,7,0,0)
bbox_5=Bbox(civils,0,6,5,241,7,0,0)
bbox_6=Bbox(civils,0,6,5,241,7,0,0)
bbox_7=Bbox(civils,0,6,5,241,7,0,0)
bbox_8=Bbox(civils,0,6,5,241,7,0,0)
bbox_9=Bbox(civils,0,6,5,241,7,0,0)
bbox_10=Bbox(civils,0,6,5,241,7,0,0)
bbox_11=Bbox(civils,0,6,5,241,7,0,0)
bbox_12=Bbox(civils,0,6,5,241,7,0,0)
bbox_13=Bbox(civils,0,6,5,241,7,0,0)
bbox_14=Bbox(civils,0,6,5,241,7,0,0)
bbox_15=Bbox(civils,0,6,5,241,7,0,0)
bbox_16=Bbox(civils,0,6,5,241,7,0,0)
bbox_17=Bbox(civils,0,6,5,241,7,0,0)
bbox_18=Bbox(civils,0,6,5,241,7,0,0)
bbox_19=Bbox(civils,0,6,5,241,7,0,0)
bbox_20=Bbox(civils,0,6,5,241,7,0,0)
....

Ground truth:
bbox_191=Bbox(civils,10.4147,17.0793,5.536,3.14,3.4553,3.792,0.0243)
bbox_192=Bbox(civils,10.367,14.0558,8.481,-0.6081,4.0335,2.8722,0.1047)
bbox_193=Bbox(civils,11.257,15.6609,8.487,-0.3036,0.1813,0.0832,0.0084)
bbox_194=Bbox(civils,9.7752,16.1657,8.4705,0.2784,2.2944,2.2734,0.0868)
bbox_195=Bbox(civils,10.4254,17.079,7.4783,0.001,2.9366,1.168,0.02)
bbox_196=Bbox(civils,13.8645,15.9638,8.1466,1.6132,2.2548,0.632,0.3322)
bbox_197=Bbox(civils,11.1839,16.3698,8.4841,1.9458,1.4746,0.801,0.0334)
bbox_198=Bbox(civils,8.5694,14.6852,8.508,1.5544,1.2779,0.3794,0.0472)
bbox_199=Bbox(civils,11.5512,15.6766,8.4876,-0.0093,0.3135,0.08,0.008)
bbox_200=Bbox(civils,8.564,16.2145,8.4918,1.5665,1.7452,0.3785,0.0349)
bbox_201=Bbox(civils,11.5213,14.7591,8.4866,2.1,0.8372,0.1815,0.0067)
bbox_202=Bbox(civils,14.5138,17.0814,6.3306,0.7263,0.0549,0.0294,0.0037)
bbox_203=Bbox(civils,10.9699,12.6984,8.4906,0.2702,0.5491,0.2056,0.0134)
bbox_204=Bbox(civils,11.6385,14.9508,8.4697,1.868,0.6006,0.3312,0.0591)
bbox_205=Bbox(civils,12.0097,15.2709,8.5033,-0.3666,0.3276,0.1802,0.0453)
bbox_206=Bbox(civils,11.1529,15.5163,8.4895,-0.5112,0.6133,0.35,0.0107)
bbox_207=Bbox(civils,6.912,16.0991,8.0578,0.0682,0.2402,0.0629,0.0029)
....

Could the problem be the scarcity of my dataset (around 50 ply scenes, between 800 and 3000 bounding boxes per scene)? Or is it a normalization problem? I understand the model was trained on completely different types of objects. I enabled do_augmentation, but not random_rotation, as I noticed no better results with random rotations.

**Comment by SamMaoYS** [comment]
Hi, thank you for exploring fine-tuning on industrial dataset. Yea, I see the data is quite different from the SpatialLM-Dataset, which are mostly indoor rooms. And as your dataset has around 50 scenes, which is limited in quantity, I think it will be quite hard to obtain good results from this dataset.
